t/lib/OpenQA/Test/FullstackUtils.pm

```diff
@@ -104,7 +104,7 @@ sub wait_for_result_panel {
             diag("stopped waiting for '$result_panel', result turned out to be '$1'");    # uncoverable statement
             return _fail_with_result_panel_contents($status_text, $msg);    # uncoverable statement
         }
-        javascript_console_has_no_warnings_or_errors;
+        ok(javascript_console_has_no_warnings_or_errors, 'No unexected js warnings');
```

Comment:
s/unexected/unexpected/

But also the problem I see that you use `ok` within a helper function scope so we don't know where this came from. I suggest to handle that with a die

Reply:
Considering this fatal here as well seems sensible, I'd rather use `_fail_with_result_panel_contents`, though, since that's how the function fails in all other cases.